File: server/Anket0roo/src/main/kotlin/com/lozanov/anket0roo/model/Questionnaire.kt
package com.lozanov.anket0roo.model

import kotlinx.serialization.Serializable
import javax.persistence.*
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Size

@Entity
@Table(name = "questionnaires")
@Serializable
data class Questionnaire(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int,

    @NotBlank
    @Size(max = 50)
    val name: String,

    val public: Boolean,

    val closed: Boolean,

    @OneToMany(
        mappedBy = "questionnaire",
        cascade = [CascadeType.ALL],
        orphanRemoval = true
    )
    val questionnaireQuestions: List<QuestionnaireQuestion>,

    @Column(name = "author_id")
    val authorId: Int

): java.io.Serializable

File: frontend/Anket0roo-App/src/model/questionnaire.ts
import {IdModel} from "./id_model";
import {Expose} from "class-transformer";
import {QuestionnaireQuestionResponse} from "./questionnaire_question_res";
import {QuestionnaireQuestion} from "./questionnaire_question";

export class Questionnaire<T extends QuestionnaireQuestion = QuestionnaireQuestionResponse> extends IdModel {
    @Expose()
    name: string;
    @Expose()
    closed: boolean;
    @Expose()
    public: boolean;
    @Expose()
    questionnaireQuestions: T[];
    @Expose()
    authorId: number;

    constructor(name: string, closed: boolean, isPublic: boolean, questionnaireQuestion: T[], authorId: number, id?: number) {
        super(id);
        this.name = name;
        this.closed = closed;
        this.public = isPublic;
        this.questionnaireQuestions = questionnaireQuestion;
        this.authorId = authorId;
    }
}